### PR TITLE
resolver: stop thread explicitly

### PR DESCRIPTION
**Pull request by BusyJay**
This pr stop the resolver thread explicitly instead of relying on drop method. When drop method is called, it's possible that the join method is called inside the resolver thread, which will lead to unexpected panic on Linux.

### CODE REVIEW

**Comment by siddontang** [comment]
PTAL @disksing

File: src/bin/tikv-server.rs [hunk 1]
```diff
@@ -953,6 +953,9 @@ fn run_raft_server(pd_client: RpcClient,
     // Stop.
     server.stop().unwrap_or_else(|err| exit_with_err(format!("{:?}", err)));
     node.stop().unwrap_or_else(|err| exit_with_err(format!("{:?}", err)));
+    if let Some(Err(e)) = worker.stop().map(|h| h.join()) {
+        info!("ignore failure when stopping resolver: {:?}", e);
```
Comment (siddontang) [inline src/bin/tikv-server.rs]: can we meet failure here?

Comment (BusyJay) [inline src/bin/tikv-server.rs]: If the thread panics, join will return error.

Comment (siddontang) [inline src/bin/tikv-server.rs]: so why not use exit_with_err directly if the thread panics?

Comment (BusyJay) [inline src/bin/tikv-server.rs]: It's unnecessary, panic hook will exit the process. And actually we don't care whether the worker exit successfully, hence just output an info log, though this line will not be visited in current case.

**Comment by siddontang** [comment]
PTAL @disksing

**Comment by disksing** [comment]
LGTM

**Pull request merged**